common/src/main/templates/io/netty/util/collection/KObjectMap.template

```diff
@@ -14,21 +14,21 @@
  */
 package io.netty.util.collection;
 
-import java.util.Collection;
+import java.util.Map;
 
 /**
  * Interface for a primitive map that uses {@code @k@}s as keys.
  *
  * @param <V> the value type stored in the map.
  */
-public interface @K@ObjectMap<V> {
+public interface @K@ObjectMap<V> extends Map<@O@, V> {
 
     /**
-     * An Entry in the map.
+     * An Entry in the map with a primitive key.
      *
      * @param <V> the value type stored in the map.
      */
-    interface Entry<V> {
+    interface PEntry<V> {
```

Comment:
@buchgr you may be right ... I don't feel strongly one way or the other.  It would certainly clean up the code to get rid of it.

@normanmaurer @Scottmitch any thoughts?